x/eibc/types/tx.go

```diff
@@ -59,13 +95,25 @@ func (m *MsgFulfillOrder) isValidOrderId(orderId string) bool {
 	return len(hashBytes) == 32
 }
 
-func (m *MsgFulfillOrder) Validate() error {
-	if err := m.ValidateBasic(); err != nil {
+func validateCommon(orderId, address, fee string) error {
+	if !isValidOrderId(orderId) {
+		return ErrInvalidOrderID
+	}
+	_, err := sdk.AccAddressFromBech32(address)
+	if err != nil {
 		return err
 	}
-	return nil
-}
 
-func (m *MsgFulfillOrder) GetFulfillerBech32Address() []byte {
-	return sdk.MustAccAddressFromBech32(m.FulfillerAddress)
+	if fee != "" {
```

Comment:
seems like if `fee=""` we pass validation

Reply:
we do, as we want to support zero fee orders
fee is `sdk.Coins` which when empty is "" string